Documentation/proposals/oci.md

```diff
@@ -0,0 +1,127 @@
+
+###### Status
+
+- The design document (https://github.com/coreos/rkt/pull/2953) is merged.
+- The implementation (https://github.com/coreos/rkt/pull/3369) is merged.
+
+###### TODOs
+
+- Introduce a dedicated remote `cimd:oci` and potentially also a local `cimd:oci-layout` (see [github.com/opencontainers/image-spec/image-layout.md](https://github.com/opencontainers/image-spec/blob/master/image-layout.md)) CIMD scheme.
+
+#### Reference based image handling
+
+The current image store implementation does not support different image formats. The blob image store only supports SHA-512.
+The ql backed SQL image store has a simple SQL scheme referencing only ACI images.
+
+In order to prepare native support for OCI the following changes need to be implemented:
+
+- Store the CIMD URI as a primary key in the current image store.
+- Support for multiple hash algorithms: Currently only SHA-512 is supported. OCI in addition needs SHA-256 and optionally SHA-384 and SHA-512.
+- The database schema needs to be reworked to reflect multi-image support.
+
+###### Status
+
+- The design and initial implementation is proposed in https://github.com/coreos/rkt/pull/3071.
+- The actual design document of the above PR can be found in [Documentation/proposals/reference_based_image_access_and_cas_store.md](https://github.com/coreos/rkt/blob/23313af1c3dac2fb24fe41f9a7c5eaca573e45dd/Documentation/proposals/reference_based_image_access_and_cas_store.md).
+
+Note that the above design document also suggests the introduction of a new key/value [Bolt](https://github.com/boltdb/bolt) based store. The current consensus is that the replacement of `ql` as the backing store can be done independently and therefore should be a non-Goal for the OCI roadmap.
+
+###### TODOs
+
+- Finalize the initial design proposal and implementation.
+
+#### Transport handlers
+
+Currently rkt directly fetches remote ACI based images or uses `docker2aci` to delegate non-ACI fetching.
+The current implementation makes it hard to integrate separate fetching subsystems due to the lack of any abstraction.
+
+###### Status
+
+The current proposal is to abstract fetching logic behind "transport handlers" allowing for independent (potentially swappable) fetching implementations for the various image formats.
+
+- A first initial design is proposed in https://github.com/coreos/rkt/pull/2964.
+- The actual design document of the above PR can be found in [Documentation/proposal/fetchers_refactor.md](https://github.com/sgotti/rkt/blob/239fdff081f9fd47dd08834a5660a1375ea4771d/Documentation/proposal/fetchers_refactor.md).
+- A first initial implementation is proposed in https://github.com/coreos/rkt/pull/3232.
+
+Note that the initial design and implementation are in very early stage and should only be considered inspirational. 
+
+###### TODOs
+
+- The current initial design proposal needs to be finalized.
```

Comment:
Fetching from disk must be supported for all formats should be referenced here or somewhere